Source/Audio/Audio/AudioManager.cpp

```diff
@@ -0,0 +1,130 @@
+#include "AudioManager.h"
+
+AudioManager::AudioManager()
+{
+
+}
+
+bool AudioManager::Init()
+{
+    if (_audioEngine != nullptr)
+        return false;
+
+    ma_engine* engine = new ma_engine();
+    ma_result result = ma_engine_init(NULL, engine);
+
+    if (result != MA_SUCCESS)
+    {
+        delete engine;
+        return false;
+    }
+
+    _audioEngine = engine;
+    return true;
+}
+
+void AudioManager::Cleanup()
+{
+    if (_audioEngine == nullptr)
+        return;
+
+    ma_engine_uninit(_audioEngine);
+
+    delete _audioEngine;
+    _audioEngine = nullptr;
+}
+
+void AudioManager::PlaySoundFile(f32 volume, bool isLooping, bool endAllOtherAudio)
+{
+    if (_audioEngine == nullptr)
+        return;
+
+    if (endAllOtherAudio && ma_sound_is_playing(_assetBrowserSound))
+        ma_sound_uninit(_assetBrowserSound);
+
+    _assetBrowserSound = new ma_sound();
+
+    ma_result result;
+    result = ma_sound_init_from_file(_audioEngine, _audioFile.c_str(), 0, NULL, NULL, _assetBrowserSound);
+    if (result != MA_SUCCESS)
+        return;
+
+    ma_sound_set_volume(_assetBrowserSound, volume);
+    ma_sound_set_looping(_assetBrowserSound, isLooping);
+    ma_sound_start(_assetBrowserSound);
+}
+
+void AudioManager::PlaySoundCommand(const std::string& fileName, f32 volume, bool isLooping, bool endAllOtherAudio)
+{
+    if (_audioEngine == nullptr)
+        return;
+
+    if (endAllOtherAudio)
+        SoundListCleanup(_consoleSounds);
+
+    ma_sound* sound = new ma_sound();
+    _consoleSounds.push_back(sound);
+
+    ma_result result;
```

Comment:
Same here, combine with line 69, no need to create this and then set it next line.